UI/LENR-app/src/app/app.component.ts

```diff
@@ -12,32 +15,72 @@ import { APP_CONSTANTS } from '../constants/app.constants';
   styleUrl: './app.component.scss'
 })
 export class AppComponent {
-  public title: string = 'LENR AI';
+  public title: string = 'LENR.ai';
   public streamText: string = '';
   public promptText: string = '';
-  public theme = APP_CONSTANTS.lightTheme;
 
-  public isFetching: boolean = false;
+  public stepText = ""
 
-  constructor() { }
+  public isPristine: boolean = false;
+  public isFetchingDocs: boolean = false;
+  public isLoading: boolean = false;
+  public hasError: boolean = false;
+  public errorMessage: string = "";
+
+  constructor(private db: DBService, private utils: UtilsService) {
+    this.isPristine = true;
+  }
+
+  get isInTranstion(): boolean {
+    return this.isFetchingDocs || this.isLoading;
+  }
+
+  get hasResult(): boolean {
+    return !this.isInTranstion && !this.isPristine && !this.hasError;
+  }
 
   getAnswer(): void {
-    console.log('Get Answer for --> ', this.promptText);
-    this.isFetching = !this.isFetching;
-    this.streamText = '';
-    this.getStream();
+    this.isPristine = false;
+    this.hasError = false;
+    this.isFetchingDocs = true;
+
+    this.db.getDocs(this.promptText).pipe(
+      map((data) => {
+        this.isFetchingDocs = false;
+        return data.prompt;
+      })).subscribe({
+        next: (prepared_prompt: string) => {
+          // this.getStream(prepared_prompt);
+          this.streamText = this.utils.processStreamData(prepared_prompt);
```

Comment:
Tested it on Vector DB data. 
Need to verify it using LLM data. (Maybe vector DB is the problem)